### DISCUSSION TITLE
Graphic showing color space

### DISCUSSION THREAD

**Discussion by claus-01** [ImageMagick/ImageMagick#2280] [Help]
Hi,

how can I generate a gradient such like this in IM?

https://res.cloudinary.com/dxaodmaf4/image/upload/v1594943464/color_space_h39dgc.png

TIA,

Claus

**Comment by snibgo** [comment]
The diagram is sometimes called the "CIE horseshoe" or "CIE tongue". The x and y axes are the x and y channels of the xyY colorspace, for some value of Y. The points on the diagram inside the shape are real colours. Those outside the shape are not real colours.

No tricolor computer screen can show all the colours inside the shape. (This is because no triangle with corners within the shape can encompass the entire shape.) So we show false colours inside the shape.  For example, we crudely convert all colours to sRGB.

I usually show x and y axes between 0.0 and 1.0.

Windows BAT syntax:
```
set HSESHOE=m 426.8 358.8 c-106.3 -106.3 -177.2 -174.2 -222.7 -219.7 -39.2 -39.2 -94 -73.4 -114.3 -73.4 -28.7 0 -39.5 45.1 -39.5 98.5 0 59.3 18.4 227.4 66 306.4 0 0 10.3 15.6 21.7 20 l 288.8 -131.8 z

set DR_HSESHOE=translate -47.8,19.2 path '%HSESHOE%'

magick ^
  -size 512x512 xc:White ^
  -sparse-color bilinear ^
0,0,#0f8,^
%%[fx:w-1],0,#ff8,^
0,%%[fx:h-1],#008,^
%%[fx:w-1],%%[fx:h-1],#f08 ^
  -set colorspace xyY ^
  -colorspace sRGB ^
  ( -size 512x512 xc:White ^
    ( +clone ^
      -draw "scale %%[fx:w/512],%%[fx:h/512] fill Black %DR_HSESHOE%" ^
    ) ^
    -alpha off -compose CopyOpacity -composite ^
  ) ^
  -compose Over -composite ^
  horseshoe4.png
```
The result is:
http://snibgo.com/imforums/horseshoe4.png

**Reply by claus-01** [reply]
Hi,

many thanks for the code. Since I am using Ubuntu, may I use your example on my [blog](https://claus-cyrny.net/wordpress/)? I have a post on color management,and I would like to use this graphic to illustrate it. I would provide a by-line & link to your site.


TIA,

Claus

**Comment by snibgo** [comment]
The code, or the image? You can share either, anywhere.
